sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/StandardMetricsExtractionProcessor.cs

```diff
@@ -0,0 +1,95 @@
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Azure.Monitor.OpenTelemetry.Exporter.Models;
+using OpenTelemetry;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Internals
+{
+    internal class StandardMetricsExtractionProcessor : BaseProcessor<Activity>
+    {
+        private bool _disposed;
+        private AzureMonitorResource _resource;
+        private readonly Meter _meter;
+        private readonly Histogram<double> _requestDuration;
+
+        internal static readonly IReadOnlyDictionary<string, string> s_standardMetricNameMapping = new Dictionary<string, string>()
+        {
+            [StandardMetricConstants.RequestDurationInstrumentName] = StandardMetricConstants.RequestDurationMetricIdValue,
+        };
+
+        internal AzureMonitorResource StandardMetricResource => _resource ??= ParentProvider.GetResource().UpdateRoleNameAndInstance();
+
+        internal StandardMetricsExtractionProcessor()
+        {
+            _meter = new Meter(StandardMetricConstants.StandardMetricMeterName);
+            _requestDuration = _meter.CreateHistogram<double>(StandardMetricConstants.RequestDurationInstrumentName);
+        }
+
+        public override void OnEnd(Activity activity)
+        {
+            if (activity.Kind == ActivityKind.Server)
+            {
+                if (_requestDuration.Enabled)
+                {
+                    activity.SetTag("_MS.ProcessedByMetricExtractors", "(Name: X,Ver:'1.1')");
+                    ReportRequestDurationMetric(activity, SemanticConventions.AttributeHttpStatusCode);
+                }
+            }
+
+            // TODO: other activity kinds
+        }
+
+        private void ReportRequestDurationMetric(Activity activity, string statusCodeAttribute)
+        {
+            string statusCodeAttributeValue = null;
+            foreach (var tag in activity.EnumerateTagObjects())
+            {
+                if (tag.Key == statusCodeAttribute)
+                {
+                    statusCodeAttributeValue = tag.Value.ToString();
+                    break;
+                }
+            }
+
+            TagList tags = default;
+            tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.RequestResultCodeKey, statusCodeAttributeValue));
+            tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.MetricIdKey, StandardMetricConstants.RequestDurationMetricIdValue));
+            tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.IsAutoCollectedKey, "True"));
+            tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.IsSyntheticKey, "False"));
+            tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.CloudRoleInstanceKey, StandardMetricResource.RoleInstance));
+            tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.CloudRoleNameKey, StandardMetricResource.RoleName));
```

Comment:
this should be in resource.

Reply:
more labels = more perf hit.
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/test/Benchmarks/Metrics/HistogramBenchmarks.cs#L35-L38

Reply:
Need to look in to this. Resource attributes are not converted to telemetry custom dimensions right now. And standard metric requires these to be at the dimension level.

Reply:
if resource attributes are not converted, then its already a bug in MetricExporter, that must be separately addressed. Atleast the rolename/roleinstance part from resource should be set as dimensions.

Reply:
> Atleast the rolename/roleinstance part from resource should be set as dimensions.

This gets converted to cloud role and role instance on telemetry level not as dimensions.

Reply:
> > Atleast the rolename/roleinstance part from resource should be set as dimensions.
> 
> This gets converted to cloud role and role instance on telemetryItem level not as dimensions.

yes got it. that is incorrect and should be fixed.